### ISSUE TITLE
Game crashes during enemy turn

### ISSUE THREAD

**Issue by RinneT** [ja2-stracciatella/ja2-stracciatella#415]
If you end the turn in the attached savegame, the game will crash after a female milita is killed.

`2016/09/06 21:15:36 ERROR   [Asserts         ] Assertion failed in Build/Tactical/PathAI.cc, line 1850
*** Program received signal SIGABRT (Aborted) ***`

For some reason iCnt becomes -128 in

```
for (iCnt=0; z != 0; iCnt++)
{
	Assert(iCnt < lengthof(guiPathingData)); // XXX TODO001C
	guiPathingData[ iCnt ] = trailTree[z].stepDir;

	z = trailTree[z].nextLink;
}
```
[CrashOnEnemyTurn.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/514625/CrashOnEnemyTurn.zip)

**Comment by derSteFfi** [comment]
-128 would be < lengthof(guiPathingData)... why is the assertion failing? and why does iCnt overflow?

**Comment by RinneT** [comment]
Two good questions. At least the second one I can now answer. iCnt is an int8, while guiPathingData is a 256 long Array.

Also immediately below the initially posted code there is giPathDataSize = (UINT8) iCnt, which can't work in this case.

It seems like iCnt is just used for everything here.

I tried making iCnt an INT32 to see what happens. But in debug it still is always an INT8.
My whole project feels a bit broken in general (debugging through code shows me wrong line numbers etc.). I have just changed to the cmake change with the Dead is Dead merge. Am I doing something wrong here? I have already thrown out the .d and .o files and the build folder to force a rebuild, but still the same?
